test/widgets/cards/history_info_card_test.dart

```diff
@@ -0,0 +1,186 @@
+    Widget createWidgetUnderTest({
+      required String date,
+      required String amount,
+    }) =>
+        ScreenUtilInit(
+          designSize: const Size(440, 956),
+          minTextAdapt: true,
+          builder: (context, child) => MaterialApp(
+            home: Material(
+              child: HistoryInfoCard(
+                date: date,
+                amount: amount,
+              ),
+            ),
+          ),
+        );
+
+    testWidgets(
+        'Given correct date and amount '
+        'when HistoryInfoCard is rendered '
+        'Then it should show the correct date and formatted amount',
+        (WidgetTester tester) async {
+      // Arrange
+      const testAmount = '1234';
+      const testDate = '12 Dec 2024';
+      const formattedTestAmount = '1,234';
+      await tester.pumpWidget(
+        createWidgetUnderTest(
+          date: testDate,
+          amount: testAmount,
+        ),
+      );
+
+      // Assert
+      expect(find.text(testDate), findsOneWidget);
+      expect(find.text('$rupeeSymbol$formattedTestAmount'), findsOneWidget);
+      expect(find.text(totalAmountText), findsOneWidget);
+    });
+
+    testWidgets(
+        'Given date and amount '
+        'when HistoryInfoCard is rendered '
+        'Then it should display a separator line between date and amount',
+        (WidgetTester tester) async {
+      // Arrange
+      const testAmount = '1234';
+      const testDate = '12 Dec 2024';
+      await tester.pumpWidget(
+        createWidgetUnderTest(
+          date: testDate,
+          amount: testAmount,
+        ),
+      );
+
+      // Assert
+      final separatorFinder = find.byType(Container).first;
+      final separatorWidget = tester.widget<Container>(separatorFinder);
+      expect(separatorWidget.decoration, isA<BoxDecoration>());
+    });
+
+    testWidgets(
+        'Given a theme with specific colors '
+        'when HistoryInfoCard is rendered '
+        'Then it should apply the correct styles from the theme',
+        (WidgetTester tester) async {
+      // Arrange
+      const testAmount = '1234';
+      const testDate = '12 Dec 2024';
+      const formattedTestAmount = '1,234';
+      await tester.pumpWidget(
+        createWidgetUnderTest(
+          date: testDate,
+          amount: testAmount,
+        ),
+      );
+
+      // Assert
+      final dateTextFinder = find.text(testDate);
+      final totalAmountTextFinder =
+          find.text('$rupeeSymbol$formattedTestAmount');
+
+      final dateTextStyle = tester.widget<Text>(dateTextFinder).style;
+      final amountTextStyle = tester.widget<Text>(totalAmountTextFinder).style;
+
+      expect(dateTextStyle?.color, ColorTokens.contrastMedium);
+      expect(amountTextStyle?.color, ColorTokens.secondary);
+    });
+
+    testWidgets(
+        'Given an empty amount '
+        'when HistoryInfoCard is rendered '
+        'Then it should display only the rupee symbol without crashing',
```

Comment:
not just rupee symbol but ₹0 should be displayed
`Then it should display 0 with rupee symbol without crashing`

Reply:
done